asserts/model.go

```diff
@@ -523,6 +523,15 @@ func (mod *Model) StorageSafety() StorageSafety {
 	return mod.storageSafety
 }
 
+// SnapdSnap returns the details of the snapd snap for this model.
+func (mod *Model) SnapdSnap() *ModelSnap {
```

Comment:
it's unclear to me if this is enough, snapd can be explicit or implicit in UC18 and UC20+ models, but seed code will pick what to use even if it's implicit:

https://github.com/snapcore/snapd/blob/2.60.4/seed/seed20.go#L699
https://github.com/snapcore/snapd/blob/2.60.4/seed/seed16.go#L321

and at least with UC20+ potentially moving from an implicit to and explict or the other way around snapd might involve a channel change but is not clear to me that this simple method plus the code in devicestate will cover those scenarios correctly?

Reply:
I have rewritten the PR so this method is not needed anymore, accounting for cases when snapd information is implicit. In the end, modelSnaps type and other types reflect only the explicit information in the model assertion, and we are not changing that now, although I think it would be good to reflect also implicit information there in the future.